app/controllers/publishers/promo_registrations_controller.rb

```diff
@@ -1,11 +1,9 @@
 class Publishers::PromoRegistrationsController < PublishersController
+  GROUP_START_DATE = Date.new(2019, 10, 1)
+
   def for_referral_code
-    promo_registration =
-      if current_publisher.admin?
-        PromoRegistration.find_by(referral_code: params[:referral_code])
-      else
-        current_publisher.promo_registrations.find_by(referral_code: params[:referral_code])
-      end
+    promo_registration = user.promo_registrations.find_by(referral_code: params[:referral_code])
```

Comment:
Use `current_user` which is defined in `ApplicationController`

Reply:
This is defined below, it's either the passed in value if the user is an administrator, or the current_user :)